### PR TITLE
pybridge: Fix superuser startup vs. package reloading race

### LINKED ISSUE #19014: "init" root pybridge gets stopped when finishing manifest reading
Trying to enable the pybridge on fedora-38 in PR #19009 creates an "interesting" bug, see [this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19009-20230626-062301-a02c31c2-fedora-38-devel-expensive/log.html#23-2). The test fails because it expects the UI to be in admin mode (the default with our test API). However, it is in limited mode.

This reproduces locally very well. To shorten the test, I did

```patch
--- test/verify/check-pages
+++ test/verify/check-pages
@@ -119,6 +119,8 @@ OnCalendar=daily
         b.enter_page("/system/services")
         b.wait_text(".service-name", "Test Service")
         b.switch_to_top()
+        b.check_superuser_indicator("Administrative access")
+        return
         self.checkDocs(["Managing services"])
         b.click("#toggle-docs")
         b.wait_visible(f'#toggle-docs + ul a:contains("Managing services")[href="{RHEL_DOC_BASE}/'
```

I then enabled running the root bridge in debugging mode. Relevant logs:

```
sudo[9285]:    admin : PWD=/ ; USER=root ; COMMAND=/usr/bin/env COCKPIT_DEBUG=all cockpit-bridge --privileged
audit[9285]: CRED_REFR pid=9285 uid=1000 auid=1000 ses=38 subj=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 msg='op=PAM:setcred grantors=pam_localuser,pam_unix acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=? re
sudo[9285]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=1000)
audit[9285]: USER_START pid=9285 uid=1000 auid=1000 ses=38 subj=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 msg='op=PAM:session_open grantors=pam_keyinit,pam_limits,pam_keyinit,pam_limits,pam_systemd,pam_unix acct="root" e
cockpit-ws[9292]: asyncio-DEBUG: Using selector: Selector
cockpit-ws[9292]: cockpit.bridge-DEBUG: Hi. How are you today?
cockpit-ws[9292]: cockpit.bridge-DEBUG: Starting the router.
cockpit-ws[9292]: cockpit.internal_endpoints-DEBUG: COCKPIT_LOGIN_MESSAGES_MEMFD wasn't set.  No login messages today.
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(/machines.d): defaulting to /machines.d
cockpit-ws[9292]: cockpit._vendor.systemd_ctypes.pathwatch-DEBUG: Starting path walk
cockpit-ws[9292]: cockpit._vendor.systemd_ctypes.pathwatch-DEBUG: r=['machines.d', '/'] dfd=-1
cockpit-ws[9292]: cockpit._vendor.systemd_ctypes.pathwatch-DEBUG: r=['machines.d'] dfd=11
cockpit-ws[9292]: cockpit._vendor.systemd_ctypes.pathwatch-DEBUG: walk ended in error 2
cockpit-ws[9292]: cockpit._vendor.systemd_ctypes.pathwatch-DEBUG: Ending existing watches.
cockpit-ws[9292]: cockpit._vendor.systemd_ctypes.pathwatch-DEBUG: Notifying of new error state 2
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(cockpit.conf): found /etc/cockpit/cockpit.conf
cockpit-ws[9292]: cockpit.config-DEBUG: cockpit.Config: loading /etc/cockpit/cockpit.conf
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(apps.override.json): defaulting to /etc/cockpit/apps.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(base1.override.json): defaulting to /etc/cockpit/base1.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(kdump.override.json): defaulting to /etc/cockpit/kdump.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(metrics.override.json): defaulting to /etc/cockpit/metrics.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(networkmanager.override.json): defaulting to /etc/cockpit/networkmanager.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(packagekit.override.json): defaulting to /etc/cockpit/packagekit.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(pcp.override.json): defaulting to /etc/cockpit/pcp.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(playground.override.json): defaulting to /etc/cockpit/playground.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(selinux.override.json): defaulting to /etc/cockpit/selinux.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(shell.override.json): defaulting to /etc/cockpit/shell.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(sosreport.override.json): defaulting to /etc/cockpit/sosreport.override.json
systemd[9269]: Created slice session.slice - User Core Session Slice.
systemd[9269]: Starting dbus-broker.service - D-Bus User Message Bus...
systemd[9269]: Started dbus-broker.service - D-Bus User Message Bus.
dbus-broker-launch[9296]: Policy to allow eavesdropping in /usr/share/dbus-1/session.conf +31: Eavesdropping is deprecated and ignored
dbus-broker-launch[9296]: Policy to allow eavesdropping in /usr/share/dbus-1/session.conf +33: Eavesdropping is deprecated and ignored
dbus-broker-launch[9296]: Ready
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(ssh.override.json): defaulting to /etc/cockpit/ssh.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(static.override.json): defaulting to /etc/cockpit/static.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(storaged.override.json): defaulting to /etc/cockpit/storaged.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(systemd.override.json): found /etc/cockpit/systemd.override.json
cockpit-ws[9292]: cockpit.config-DEBUG: lookup_config(users.override.json): defaulting to /etc/cockpit/users.override.json
cockpit-ws[9292]: cockpit.superuser-DEBUG: set_configs() with 7 items
cockpit-ws[9292]: cockpit.superuser-DEBUG:   bridges are now ['sudo', 'pkexec']
cockpit-ws[9292]: cockpit.transports-DEBUG: Created transport <cockpit.transports.StdioTransport object at 0x7f5b0e8e4710> for protocol <cockpit.bridge.Bridge object at 0x7f5b0e89ba90>, fds 0 1
cockpit-ws[9292]: cockpit.protocol-DEBUG: connection_made(<cockpit.transports.StdioTransport object at 0x7f5b0e8e4710>)
cockpit-ws[9292]: cockpit.protocol-DEBUG: sending message  {'command': 'init', 'version': 1, 'checksum': '850ddd57f79491d0f7b3741ca9c113b1632a6f821f06de793d6b1c33fa57f227', 'packages': {'apps': None, 'base1': None, 'kdump': None, 'm
cockpit-ws[9292]: cockpit.protocol-DEBUG: writing to transport <cockpit.transports.StdioTransport object at 0x7f5b0e8e4710>
cockpit-ws[9292]: cockpit.bridge-DEBUG: Startup done.  Looping until connection closes.
cockpit-ws[9292]: cockpit.transports-DEBUG: Read ready on <cockpit.transports.StdioTransport object at 0x7f5b0e8e4710> <cockpit.bridge.Bridge object at 0x7f5b0e89ba90> 0
cockpit-ws[9292]: cockpit.transports-DEBUG:   got EOF
cockpit-ws[9292]: cockpit.protocol-DEBUG: connection_lost
sudo[9285]: pam_unix(sudo:session): session closed for user root
```

There are no tracebacks, errors, or warnings. I suppose the development mode changes the timing, as it has to hash a lot bigger files. But this feels like a more fundamental race condition?

### PR DESCRIPTION
**Pull request by martinpitt**
When the superuser bridge gets auto-started via the init message or the
integration tests, it can happen that packages get reloaded while the
superuser bridge already runs, but is still initializing. At that point,
`self.current` is still "init", and once packages finished loading,
set_config() stopped that "init" bridge as it is not decleared by the
manifests (usually "sudo" and "pkexec"). That led to the session
sometimes not being privileged.

This was particularly prone to happen in the "devel" scenario where the
packages are much larger and processing took much longer.

We need to keep the "init" phase for correct error tracking, and that
value is also recognized by the shell to hide the indicator. Avoid
the race by remembering the current peer name separately as a new member
of SuperuserPeer.

Fixes #19014

----

Broken out from PR #19009 to fix [these errors](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19009-20230626-062301-a02c31c2-fedora-38-devel-expensive/log.html). They happen in the /devel scenario due to timing issues, but are in principle a bug with production bridge.

### CODE REVIEW

File: src/cockpit/superuser.py [hunk 1]
```diff
@@ -168,7 +168,7 @@ async def go(self, name: str, responder: ferny.InteractionResponder) -> None:
         except KeyError as exc:
             raise bus.BusError('cockpit.Superuser.Error', f'Unknown superuser bridge type "{name}"') from exc
 
-        self.current = 'init'
+        self.current = f'init-{name}'
```
Comment (allisonkarlitskaya) [inline src/cockpit/superuser.py:171]: The value "init" is special.  Look in `shell/superuser.jsx`:

```js
    const show = superuser_proxy.Current != "root" && superuser_proxy.Current != "init";
```
```

Comment (martinpitt) [reply src/cockpit/superuser.py:171]: lis | or maybe like, the peer object is already there, no? you could ask it its name

Comment (martinpitt) [reply src/cockpit/superuser.py]: Turns out `SuperuserPeer` does not actually know its own name, so I added a new `self.peer_name` state. This is a bit more unwieldy, but avoids changing the `current` D-Bus property. I re-tested this with the devel scenario, and it works fine.

**Review by mvollmer — APPROVED** [review]
Thanks, looks good now.

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]
I'd have a strong preference to see the name passed through to `SuperuserPeer.__init__` and hung on that object, instead.

There's already *way* too much state on `SuperuserRoutingRule` and we don't do a very good job of documenting or defining invariants on it (at all)...

**Comment by martinpitt** [comment]
@allisonkarlitskaya Ack -- I considered that as well and thought it wouldn't quite belong there as SuperuserPeer does not need to know its name. But it's cleaner indeed. Reworked, and re-tested in the context of #19009

**Review by allisonkarlitskaya — APPROVED** [review]
Love it!  Thanks.

**Pull request merged**